### PR TITLE
Issue #9143:AvoidEscapedUnicodeCharacters does not identify line ending characters in multiline text blocks

### LINKED ISSUE #9143: AvoidEscapedUnicodeCharacters should support trailing '\' escape symbol
https://checkstyle.org/config_misc.html#AvoidEscapedUnicodeCharacters

From https://github.com/checkstyle/checkstyle/pull/9135#discussion_r550064696:

>The docs says there is one more escape - for a trailing \. Something like \\$.

The above comment is referencing this doc: https://docs.oracle.com/javase/specs/jls/se15/html/jls-3.html#jls-3.10.7

_______________________________________________________

```
➜  src cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
        <module name="AvoidEscapedUnicodeCharacters">
            <property name="allowIfAllCharactersEscaped" value="true"/>
        </module>
    </module>
</module>

➜  src cat TestClass.java 
public class TestClass {
    static final String value1 = """
        \u03bc\
        \s\u03bc\
        """; // should be no violation on line 2
}

➜  src javac TestClass.java 

➜  src java -jar checkstyle-8.39-SNAPSHOT-all.jar -c config.xml TestClass.java
Starting audit...
[ERROR] src/TestClass.java:2:37: Unicode escape(s) usage should be avoided. [AvoidEscapedUnicodeCharacters]
Audit done.
Checkstyle ends with 1 errors.

```
AvoidEscapedUnicodeCharacters should not report a violation in this case, since all characters are escaped in this text block (`value1`).

### PR DESCRIPTION
**Pull request by shashwatj07**
Resolves #9143

Diff Regression config: https://gist.githubusercontent.com/shashwatj07/62612965266474738d91ed8a1eda53fd/raw/06c9dc513f0fa2e8086aa1e1d76abd7491458a53/config1.xml
Diff Regression projects: https://gist.githubusercontent.com/shashwatj07/b3de11a5c370e5f6092e31776b9c3b91/raw/8f42d25090f9d1b10c9b5d56ba39b482ac0aa91f/projects.properties

### CODE REVIEW

**Comment by nmancus1** [comment]
Please generate a check regression report, I would use all configs from the documentation: https://checkstyle.sourceforge.io/config_misc.html#AvoidEscapedUnicodeCharacters . Make sure to use the `id` property to differentiate between violations.

**Comment by shashwatj07** [comment]
Github, generate report

**Comment by nmancus1** [comment]
We have fixed the issue with report generation project file, you should be able to just generate the report with default action (supplying only a config).

**Comment by shashwatj07** [comment]
Github, generate report

**Comment by nmancus1** [comment]
`spring-integration` has changed from `master` to `main`, this is the reason for report failure.

**Comment by shashwatj07** [comment]
Github, generate report

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/AvoidEscapedUnicodeCharactersCheck.java [hunk 1]
```diff
@@ -208,6 +208,7 @@
             + "|\\\\b"
             + "|\\\\f"
             + "|\\\\n"
+            + "|\\R"
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/AvoidEscapedUnicodeCharactersCheck.java:211]: https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html#lineending

**Review by romani — APPROVED** [review]
looks good

**Pull request merged**